lib/ElementHandle.js

```diff
@@ -14,46 +14,40 @@
  * limitations under the License.
  */
 const path = require('path');
+const {JSHandle} = require('./ExecutionContext');
 const {helper} = require('./helper');
 
-class ElementHandle {
+class ElementHandle extends JSHandle {
   /**
-   * @param {!Frame} frame
-   * @param {!Connection} client
+   * @param {!ExecutionContext} context
+   * @param {!Session} client
    * @param {!Object} remoteObject
    * @param {!Mouse} mouse
    * @param {!Touchscreen} touchscreen;
    */
-  constructor(frame, client, remoteObject, mouse, touchscreen) {
-    this._frame = frame;
-    this._client = client;
-    this._remoteObject = remoteObject;
+  constructor(context, client, remoteObject, mouse, touchscreen) {
+    super(context, client, remoteObject);
     this._mouse = mouse;
     this._touchscreen = touchscreen;
-    this._disposed = false;
   }
 
   /**
-   * @return {?string}
+   * @override
+   * @return {?ElementHandle}
    */
-  _remoteObjectId() {
-    return this._disposed ? null : this._remoteObject.objectId;
-  }
-
-  async dispose() {
-    if (this._disposed)
-      return;
-    this._disposed = true;
-    await helper.releaseObject(this._client, this._remoteObject);
+  asElement() {
+    return this;
   }
 
   /**
    * @return {!Promise<{x: number, y: number}>}
    */
   async _visibleCenter() {
-    const center = await this._frame.evaluate(element => {
+    const center = await this.executionContext().evaluate(element => {
       if (!element.ownerDocument.contains(element))
         return null;
+      if (element.nodeType !== HTMLElement.ELEMENT_NODE)
```

Comment:
Why are we making element handles for nodes?

Reply:
There's no good way to differentiate for now;  as discussed offline, I'll make them throw for elementHandle methods for now.